commerce_coordinator/apps/frontend_app_ecommerce/views.py

```diff
@@ -81,17 +83,37 @@ def get(self, request):
         """Return paginated response of user's order history."""
         user = request.user
         user.add_lms_user_id("UserOrdersView GET method")
+
         # build parameters
         params = {
-            'customer_id': request.query_params.get('customer_id'),
-            'username': request.user.username,
-            'email': request.user.email,
-            "edx_lms_user_id": request.user.lms_user_id,
+            "customer_id": None,
+            "username": None,
+            "email": None,
+            "edx_lms_user_id": None,
             "page": 0,
             "page_size": ORDER_HISTORY_PER_SYSTEM_REQ_LIMIT,
             "cutoff_in_days": ORDER_HISTORY_PER_SYSTEM_REQ_CUTOFF_IN_DAYS,
         }
 
+        # If staff, allow override of user from query params
+        learner_username = request.query_params.get("username")
+        if request.user.is_staff and learner_username:
+            params["username"] = learner_username
+            try:
+                learner = User.objects.get(username=learner_username)
+                params["email"] = learner.email
+                params["edx_lms_user_id"] = learner.lms_user_id
+            except User.DoesNotExist:
+                logger.info(
+                    "[UserOrdersView] User does not exist in Coordinator "
+                    "and hence doesn't have any Commercetools orders either"
```

Comment:
This is not true, actually, a CT user would always exist if that user has an order. There are a few scenarios now where the user does not exist in CC, but the order was created on CT. This was due to direct upselling URL to CT from the marketing site.